File: app/api/shared/create.js
const fs = require('fs');
const path = require('path');
const usersFilePath = path.join(process.cwd(), 'storage', 'db', 'users.json');
const staticFilePath = path.join(process.cwd(), 'storage', 'static', 'static.json');
const readUsers = () => {
    if (fs.existsSync(usersFilePath)) {
        const data = fs.readFileSync(usersFilePath, 'utf8');
        return JSON.parse(data) || [];
    }
    return [];
};
const readStatic = () => {
    if (fs.existsSync(staticFilePath)) {
        const data = fs.readFileSync(staticFilePath, 'utf8');
        return JSON.parse(data) || [];
    }
    return [];
};
const writeStatic = (data) => {
    fs.writeFileSync(staticFilePath, JSON.stringify(data, null, 2));
};
const createFolderLogic = (req, res) => {
    const { accessToken, folderName, id } = req.body;
    if (!accessToken || !folderName || !id) {
        return res.status(400).send('Access token, folder name, and user ID are required');
    }
    const users = readUsers();
    const user = users.find(u => u.accessToken === accessToken && u.id === id);
    if (!user) {
        return res.status(401).send('Invalid access token or user ID');
    }
    const folderPath = path.join(process.cwd(), 'storage', 'static', 'storage', folderName);
    fs.mkdir(folderPath, { recursive: true }, (err) => {
        if (err) {
            return res.status(500).send('Error creating folder');
        }
        const staticData = readStatic();
        staticData.push({ folderName, createdBy: user.username, createdAt: new Date(), userId: user.id });
        writeStatic(staticData);
        return res.status(201).send(`Folder "${folderName}" created successfully`);
    });
};
module.exports = { createFolderLogic };

File: app/api/shared/create.py
import requests

def create_folder(access_token, folder_name, user_id):
    url = "http://localhost:80/create/shared"
    headers = {
        "Content-Type": "application/json"
    }
    payload = {
        "accessToken": access_token,
        "folderName": folder_name,
        "id": user_id
    }
    try:
        response = requests.post(url, json=payload, headers=headers)
        if response.status_code == 201:
            print(f'Success: {response.text}')
        elif response.status_code == 400:
            print('Error: Access token, folder name, and user ID are required.')
        elif response.status_code == 401:
            print('Error: Invalid access token or user ID.')
        elif response.status_code == 500:
            print('Error: An error occurred while creating the folder.')
        else:
            print(f'Unexpected error: {response.status_code} - {response.text}')
    except requests.exceptions.RequestException as e:
        print(f'An error occurred while making the request: {e}')
if __name__ == "__main__":
    access_token = "tokenaro-M9KBkenjmaSa0wmOV0CkuU3oZmX2xcLxQpyeRngR" 
    folder_name = "test123" 
    user_id = 1
    create_folder(access_token, folder_name, user_id)